### PR TITLE
added financial review agent

### PR DESCRIPTION
**Pull request by HussainKhanSherwani**
# Pull Request Template

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

### Code Review
- [ ] I have carefully reviewed my code and believe it is final from my side.
- [ ] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [ ] I have tested the new feature or updates in both light and dark themes.
- [ ] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [ ] I have tested my changes on mobile screen sizes.
- [ ] I have tested my changes on tablet screen sizes.
- [ ] I have tested my changes on laptop screen sizes.

### Functional Testing
- [ ] I have thoroughly tested the functionality of my changes.
- [ ] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

- Example test 1: Tested feature X by uploading the image
- Example test 2: Tested feature X by doing these things.

# Demo Video
Please provide a demo video of your changes. This helps reviewers understand the changes and verify the functionality.

You can record the video using tools like Loom(https://www.loom.com/), Clip by ClickUp(https://clickup.com/features/clips), Sendspark(https://www.sendspark.com/), etc.

Include the link to the video below and make sure the video covers
- [ ] Explain the task/bug you are solving
- [ ] Show the changes in the UI
- [ ] Show the changes in the codebase
- [ ] Show the changes you tested to make sure it works
- [ ] Show the changes in various screen sizes
- [ ] Show the changes in both light and dark themes

### CODE REVIEW

File: dodao-ai-agents/langgraph-poetry/langgraph_poetry/financial_review_agent.py [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+    scraped_content = state["scraped_content"]
+    all_metrics = {}
+    relevant_metrics = {}
+
+    for link, content in scraped_content.items():
+        if "Failed to scrape" in content:
+            continue
+
+        prompt = (
+            "From the content provided, extract the following details:\n"
```
Comment (RobinNagpal) [inline dodao-ai-agents/langgraph-poetry/langgraph_poetry/financial_review_agent.py:153]: Any relevance of the years and the date the information was captured?

File: dodao-ai-agents/langgraph-poetry/langgraph_poetry/financial_review_agent.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+
+    if additional_data["relevant_metrics"]:
+        for key, value in additional_data["relevant_metrics"].items():
+            prompt += f"- **{key}:** {value}\n"
+    else:
+        prompt += "- No additional relevant metrics were found.\n"
+
+    prompt += "\nPrepare a comprehensive report that highlights the key financial information of the startup that a potential investor must be aware of before deciding to invest in this startup based on:\n"
+    prompt += (
+        "- An **Executive Summary** summarizing the startup's financial health and key takeaways.\n"
```
Comment (RobinNagpal) [inline dodao-ai-agents/langgraph-poetry/langgraph_poetry/financial_review_agent.py:262]: Things that we want to analyze

1) For this sector, what does a normal financial outlook look for a startup? For example how much spending is needed, when can the startup turn profitable in this sector, what are the major expenses overall.

--- Do this in a separate LLM call ---
2) Generic feedback on the financial outlook of the company on each metric, This can be additional column.

--- Do this in a separate LLM call ---
3)  Sector specific information on each of the metrics of the financial information. This can be additional column
